contrib/inventory/brook.py

```diff
@@ -0,0 +1,262 @@
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Brook.io external inventory script
+==================================
+
+Generates inventory that Ansible can understand by making API requests to Brook.io via the libbrook
+library. Hence, such dependency must be installed in the system to run this script.
+
+The default configuration file is named 'brook.ini' and is located alongside this script. You can
+choose any other file by setting the BROOK_INI_PATH environment variable.
+
+If param 'project_id' is left blank in 'brook.ini', the inventory includes all the instances in
+projects where the requesting user belongs. Otherwise, only instances from the given project are
+included, provided the requesting user belongs to it.
+
+The following variables are established for every host. They can be retrieved from the hostvars
+dictionary.
+ - brook_name: str
+ - brook_description: str
+ - brook_project: str
+ - brook_template: str
+ - brook_region: str
+ - brook_status: str
+ - brook_tags: list(str)
+ - brook_internal_ips: list(str)
+ - brook_external_ips: list(str)
+ - brook_created_at
+ - brook_updated_at
+ - ansible_ssh_host
+
+Instances are grouped by the following categories:
+ - tag:
+   A group is created for each tag. E.g. groups 'tag_foo' and 'tag_bar' are created if there exist
+   instances with tags 'foo' and/or 'bar'.
+ - project:
+   A group is created for each project. E.g. group 'project_test' is created if a project named
+   'test' exist.
+ - status:
+   A group is created for each instance state. E.g. groups 'status_RUNNING' and 'status_PENDING'
+   are created if there are instances in running and pending state.
+
+Examples:
+  Execute uname on all instances in project 'test'
+  $ ansible -i brook.py project_test -m shell -a "/bin/uname -a"
+
+  Install nginx on all debian web servers tagged with 'www'
+  $ ansible -i brook.py tag_www -m apt -a "name=nginx state=present"
+
+  Run site.yml playbook on web servers
+  $ ansible-playbook -i brook.py site.yml -l tag_www
+
+Support:
+  This script is tested on Python 2.7 and 3.4. It may work on other versions though.
+
+Author: Francisco Ros <fjros@doalitic.com>
+Version: 0.1
+"""
+
+
+import sys
+import os
+
+try:
+    from ConfigParser import SafeConfigParser as ConfigParser
+except ImportError:
+    from configparser import ConfigParser
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+try:
+    import libbrook
+except:
+    print('Brook.io inventory script requires libbrook. See https://github.com/doalitic/libbrook')
+    sys.exit(1)
```

Comment:
errors should go to stderr, you can use `sys.exit("msg") ` to do both in one statement